<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>770</width>
    <height>844</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NDS Factory</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="maximumSize">
    <size>
     <width>756</width>
     <height>800</height>
    </size>
   </property>
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="sizeConstraint">
     <enum>QLayout::SetMinAndMaxSize</enum>
    </property>
    <item>
     <widget class="QTabWidget" name="mainTab">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="unpackerTab">
       <attribute name="title">
        <string>Unpacker</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Rom</string>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>4</number>
           </property>
           <property name="sizeConstraint">
            <enum>QLayout::SetMinimumSize</enum>
           </property>
           <property name="leftMargin">
            <number>1</number>
           </property>
           <property name="topMargin">
            <number>1</number>
           </property>
           <property name="rightMargin">
            <number>1</number>
           </property>
           <property name="bottomMargin">
            <number>1</number>
           </property>
           <item>
            <widget class="QLineEdit" name="loadedRomPath">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="loadRomBtn">
             <property name="text">
              <string>Load Rom</string>
             </property>
             <property name="autoDefault">
              <bool>false</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Header</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <widget class="QTableView" name="unpackerHeaderDataTable">
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustToContents</enum>
             </property>
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="wordWrap">
              <bool>false</bool>
             </property>
             <attribute name="horizontalHeaderVisible">
              <bool>false</bool>
             </attribute>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_3">
          <property name="title">
           <string>Tools</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QGroupBox" name="unpackerExtractorGbx">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="title">
              <string>Extractor</string>
             </property>
             <layout class="QVBoxLayout" name="verticalLayout_4">
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_2">
                <property name="leftMargin">
                 <number>7</number>
                </property>
                <property name="topMargin">
                 <number>7</number>
                </property>
                <property name="rightMargin">
                 <number>7</number>
                </property>
                <property name="bottomMargin">
                 <number>7</number>
                </property>
                <item>
                 <widget class="QPushButton" name="unpackerDumpHeaderBtn">
                  <property name="text">
                   <string>Header</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpArm9Btn">
                  <property name="text">
                   <string>ARM9 Bin</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpArm7Btn">
                  <property name="text">
                   <string>ARM7 Bin</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpFntBtn">
                  <property name="text">
                   <string>Filename Table</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_3">
                <property name="leftMargin">
                 <number>7</number>
                </property>
                <property name="topMargin">
                 <number>7</number>
                </property>
                <property name="rightMargin">
                 <number>7</number>
                </property>
                <property name="bottomMargin">
                 <number>7</number>
                </property>
                <item>
                 <widget class="QPushButton" name="unpackerDumpFatBtn">
                  <property name="text">
                   <string>FAT</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpFatFilesBtn">
                  <property name="text">
                   <string>FAT Files</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpArm9OverlayBtn">
                  <property name="text">
                   <string>ARM9 Overlay</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpArm9OverlayFilesBtn">
                  <property name="text">
                   <string>ARM9 Overlay Files</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
              <item>
               <layout class="QHBoxLayout" name="horizontalLayout_7">
                <property name="leftMargin">
                 <number>7</number>
                </property>
                <property name="topMargin">
                 <number>7</number>
                </property>
                <property name="rightMargin">
                 <number>7</number>
                </property>
                <property name="bottomMargin">
                 <number>7</number>
                </property>
                <item>
                 <widget class="QPushButton" name="unpackerDumpArm7OverlayBtn">
                  <property name="text">
                   <string>ARM7 Overlay</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpArm7OverlayFilesBtn">
                  <property name="text">
                   <string>ARM7 Overlay Files</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="unpackerDumpIconTitleLogoBtn">
                  <property name="text">
                   <string>Icon/Title Logo</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QGroupBox" name="unpackerExtraGbx">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="title">
              <string>Extra</string>
             </property>
             <layout class="QHBoxLayout" name="horizontalLayout_4">
              <item>
               <widget class="QPushButton" name="unpackerDumpEverythingBtn">
                <property name="text">
                 <string>Extract Everything</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="unpackerDecodeFatFilesBtn">
                <property name="text">
                 <string>Decode FAT Contents</string>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="packerTab">
       <attribute name="title">
        <string>Packer</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item>
         <widget class="QGroupBox" name="groupBox_4">
          <property name="title">
           <string>Header</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_6">
           <item>
            <widget class="QTableView" name="packerHeaderDataTable">
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustToContents</enum>
             </property>
             <property name="editTriggers">
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_33">
             <property name="sizeConstraint">
              <enum>QLayout::SetMinimumSize</enum>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="packerCalcHeaderCrcBtn">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Calc Header CRC</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadHeaderBtn">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="text">
                <string>Import Header From File</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="unpackerInjectorGbx">
          <property name="title">
           <string>Injector</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_26">
             <item>
              <widget class="QLineEdit" name="loadedArm9BinPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadArm9BinBtn">
               <property name="text">
                <string>Load Arm9 Bin</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="loadedArm7BinPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadArm7BinBtn">
               <property name="text">
                <string>Load Arm7 Bin</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_28">
             <item>
              <widget class="QLineEdit" name="loadedFntPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadFntBtn">
               <property name="text">
                <string>Load FNT</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="loadedFatPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadFatBtn">
               <property name="text">
                <string>Load FAT</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_29">
             <item>
              <widget class="QLineEdit" name="loadedArm9OverlayPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadArm9OverlayBtn">
               <property name="text">
                <string>Load ARM9 Overlay</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="loadedArm9OverlayFilesPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadArm9OverlayFilesBtn">
               <property name="text">
                <string>Load ARM9 Overlay Files</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_30">
             <item>
              <widget class="QLineEdit" name="loadedArm7OverlayPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadArm7OverlayBtn">
               <property name="text">
                <string>Load ARM7 Overlay</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="loadedArm7OverlayFilesPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadArm7OverlayFilesBtn">
               <property name="text">
                <string>Load ARM7 Overlay Files</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_31">
             <item>
              <widget class="QLineEdit" name="loadedIconTitlePathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadIconTitleBtn">
               <property name="text">
                <string>Load Icon/Title</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="loadedFatFilesPathEdt"/>
             </item>
             <item>
              <widget class="QPushButton" name="packerLoadFatFilesBtn">
               <property name="text">
                <string>Load FAT Files</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_5">
          <property name="layoutDirection">
           <enum>Qt::LeftToRight</enum>
          </property>
          <property name="autoFillBackground">
           <bool>false</bool>
          </property>
          <property name="title">
           <string>Builder</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_10">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_32">
             <property name="sizeConstraint">
              <enum>QLayout::SetDefaultConstraint</enum>
             </property>
             <item>
              <widget class="QRadioButton" name="packerPadType00RdBtn">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>PAD Sections With 00s</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="packerPadTypeFFRdBtn">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="text">
                <string>PAD Sections With FFs</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_11">
             <item>
              <widget class="QCheckBox" name="packerTrimRomsCbx">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="text">
                <string>Generate Trimmed Rom</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_9">
             <item>
              <widget class="QPushButton" name="packerBuildNDSRomBtn">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>Build NDS Rom</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="fatPatchingTab">
       <attribute name="title">
        <string>Fat Patching</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_13">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_14">
          <property name="spacing">
           <number>4</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="label_83">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>If you FAT Files Address (fat_data.bin) is different from the original ROM position, you have to patch the FAT (fat.bin)...</string>
            </property>
            <property name="scaledContents">
             <bool>false</bool>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_12">
          <item>
           <widget class="QGroupBox" name="groupBox_6">
            <property name="title">
             <string>FAT Patcher</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_15">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_35">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLineEdit" name="fatPatchingFatPathEdt"/>
               </item>
               <item>
                <widget class="QPushButton" name="fatPatchingLoadFatBtn">
                 <property name="text">
                  <string>Load FAT (fat.bin)</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_36">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QLabel" name="label_84">
                 <property name="text">
                  <string>Original Fat Files (fat_data.bin) Addr:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="fatPatchingOriginalFatFilesAddrEdt"/>
               </item>
               <item>
                <widget class="QLabel" name="label_85">
                 <property name="text">
                  <string>New Fat Files Addr:</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLineEdit" name="fatPatchingNewFatFilesAddrEdt"/>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_37">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QPushButton" name="fatPatchingPatchFatBtn">
                 <property name="text">
                  <string>Apply Patch!</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>770</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="enabled">
    <bool>true</bool>
   </property>
  </widget>
  <action name="actionAbout">
   <property name="text">
    <string>About</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
